Criterion/Solr/Search/Query/CriterionVisitor/PublishedOnDayOfWeek.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\CookbookBundle\Criterion\Solr\Search\Query\CriterionVisitor;
+
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use EzSystems\EzPlatformSolrSearchEngine\Query\CriterionVisitor;
+use EzSystems\CookbookBundle\Criterion as CustomCriterion;
+
+class PublishedOnDayOfWeek extends CriterionVisitor
+{
+    /**
+     * CHeck if visitor is applicable to current criterion.
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\Query\Criterion $criterion
+     *
+     * @return bool
+     */
+    public function canVisit(Criterion $criterion)
+    {
+        return $criterion instanceof CustomCriterion\PublishedOnDayOfWeek;
+    }
+
+    /**
+     * Map field value to a proper Solr representation.
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\Query\Criterion $criterion
+     * @param \EzSystems\EzPlatformSolrSearchEngine\Query\CriterionVisitor $subVisitor
+     *
+     * @return string
+     */
+    public function visit(Criterion $criterion, CriterionVisitor $subVisitor = null)
+    {
+        return implode(
+            ' OR ',
+            array_map(
+                function ($value) {
+                    // @todo: find a way to filter day of week here
+                    return 'content_publication_date_dt:""';
```

Comment:
This is the part where I'm stuck. `$value` holds integer corresponding to a day of week (0 = Sun... 6 = Sat). I tried to search how to query for this using Solr, but can't find any relevant info.
Any ideas @andrerom, @pspanja?